mailpoet/assets/js/src/automation/editor/store/actions.ts

```diff
@@ -54,12 +54,12 @@ export function* save() {
   const data = yield apiFetch({
     path: `/workflows/${workflow.id}`,
     method: 'PUT',
-    data: workflow,
+    data: { ...workflow },
   });
 
   return {
     type: 'SAVE',
-    workflow: data.data,
+    workflow: data?.data ?? workflow,
```

Comment:
It can occur when the error handler implemented in https://github.com/mailpoet/mailpoet/pull/4359/commits/e31f7d66ffad8c4537b25fafc064db0fe3cabe52 catches an error and displays a notice. In such a case, I believe using the original workflow is probably correct because we were not able to save the workflow for some reason. The result will be something like this (still rendering the original workflow):

<img width="1516" alt="Screen Shot 2022-09-12 at 7 13 00" src="https://user-images.githubusercontent.com/141436/189578242-87daf1d7-b123-4a02-9d68-cd601449891f.png">

Reply:
That makes perfect sense, thanks for the explanation!